### DISCUSSION TITLE
Please add Material 3

### DISCUSSION THREAD

**Discussion by App-Developer-Ayush** [Sketchware-Pro/Sketchware-Pro#460] [Ideas]
Please add Material 3 components and all the Material 3 things. So I request you to add this in the next update. I have bored using Material 2 and it's stuff. And after adding Material 3, add an option in project settings to select Material 2 or Material 3 for the project.

**Comment by j-px** [comment]
We won't add new Views or Components, as you can create them on your own, but we might add a toggle/Spinner to Project Configuration to easily set the application theme to be Material 3.

**Comment by App-Developer-Ayush** [comment]
Ok, I will wait for that update.

**Comment by j-px** [comment]
To enable it without a toggle, first enable the new MaterialComponents theme in Project Configuration.  
Then use an XML Command Block to replace 3 text snippets in styles.xml:
 - `"Theme.MaterialComponents.Light.NoActionBar"` with `"Theme.Material3.Light.NoActionBar"`
 - `"ThemeOverlay.MaterialComponents.Dark.ActionBar"` with `"ThemeOverlay.Material3.Dark"`
 - `"ThemeOverlay.MaterialComponents.Light"` with `"ThemeOverlay.Material3.Light"`

**Reply by DevVigilante** [reply]
> To enable it without a toggle, first enable the new MaterialComponents theme in Project Configuration. Then use an XML Command Block to replace 3 text snippets in styles.xml:
> 
> * `"Theme.MaterialComponents.Light.NoActionBar"` with `"Theme.Material3.Light.NoActionBar"`
> * `"ThemeOverlay.MaterialComponents.Dark.ActionBar"` with `"ThemeOverlay.Material3.Dark"`
> * `"ThemeOverlay.MaterialComponents.Light"` with `"ThemeOverlay.Material3.Light"`

Instead of doing this hard thing you can create styles.xml and colors.xml in resource manager.This will replace built in styles.xml and colors.xml with new one while compiling.

**Comment by srt54558** [comment]
> To enable it without a toggle, first enable the new MaterialComponents theme in Project Configuration.  
> Then use an XML Command Block to replace 3 text snippets in styles.xml:
>  - `"Theme.MaterialComponents.Light.NoActionBar"` with `"Theme.Material3.Light.NoActionBar"`
>  - `"ThemeOverlay.MaterialComponents.Dark.ActionBar"` with `"ThemeOverlay.Material3.Dark"`
>  - `"ThemeOverlay.MaterialComponents.Light"` with `"ThemeOverlay.Material3.Light"`

Where can I  use XML command blocks?

**Reply by j-px** [reply]
Enable them first by enabling showing built-in Blocks in Mod Settings, then you can find them in the More Block palette while editing an Event.
